File: module/SampleModule/src/main/java/com/pds/sample/module/plugin/MainActivity.java
package com.pds.sample.module.plugin;

import android.app.Activity;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.os.Handler;
import android.os.RemoteException;
import android.view.View;
import android.widget.Toast;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.pds.plugin.pm.PluginManager;
import com.pds.sample.R;

import java.io.File;


public class MainActivity extends AppCompatActivity {
    private Handler mHandler = new Handler();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.layout);

    }

    public Activity getActivity() {
        return this;
    }

    private void doInstall(final File apkPath) {
        try {
            final PackageInfo info = getPackageManager().getPackageArchiveInfo(apkPath.getAbsolutePath(), PackageManager.GET_ACTIVITIES);
            if (info == null) {
                mHandler.post(new Runnable() {
                    @Override
                    public void run() {
                        Toast.makeText(getActivity(), "apk损坏\n" + apkPath.getAbsolutePath(), Toast.LENGTH_SHORT).show();
                    }
                });
                return;
            }
            final int re = PluginManager.getInstance().installPackage(apkPath.getAbsolutePath(), 0);
        } catch (RemoteException e) {
            e.printStackTrace();
        }
    }

    public void load(View view) {
    }
}

File: study/sdk_use/src/main/java/com/pds/sdk/use/MainActivity.kt
package com.pds.sdk.use

import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.pds.sdk.use.activity.ActivityLibUse
import org.jetbrains.anko.internals.AnkoInternals


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun doClick(view: View) {
        AnkoInternals.internalStartActivity(this,ActivityLibUse::class.java, emptyArray())
    }

}